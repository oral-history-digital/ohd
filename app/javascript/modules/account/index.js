export { NAME as ACCOUNT_NAME } from './constants';

export { LOGOUT } from './action-types';

export { default as accountReducer } from './reducer';

export { getIsLoggedIn, getIsLoggedOut, getLoggedInAt, getCheckedOhdSession } from './selectors';

export { default as AccountPage } from './components/AccountPage';
export { default as AccountContainer } from './components/AccountContainer';
export { default as OrderNewPasswordContainer } from './components/OrderNewPasswordContainer';
export { default as RegisterContainer } from './components/RegisterContainer';
export { default as ActivateAccount } from './components/ActivateAccount';
export { default as RedirectOnLogin } from './components/RedirectOnLogin';
