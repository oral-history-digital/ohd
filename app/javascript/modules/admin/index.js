export { default as ArchivePage } from './ArchivePage';
export { default as CollectionsSearchFormContainer } from './CollectionsSearchFormContainer';
export { default as ContributionTypesSearchFormContainer } from './ContributionTypesSearchFormContainer';
export { default as DataList } from './DataList';
export { default as EditInterview } from './EditInterview';
export { default as EditProjectConfig } from './EditProjectConfig';
export { default as EditProjectAccessConfig } from './EditProjectAccessConfig';
export { default as EditProjectDisplay } from './EditProjectDisplay';
export { default as EditProjectInfo } from './EditProjectInfo';
export { default as EditViewOrRedirect } from './EditViewOrRedirect';
export { default as EventTypesAdminPage } from './EventTypesAdminPage';
export { default as ExternalLinksContainer } from './ExternalLinksContainer';
export { default as HelpTextAdminPage } from './HelpTextAdminPage';
export { default as UsersAdminPage } from './UsersAdminPage';
export { default as LanguagesSearchFormContainer } from './LanguagesSearchFormContainer';
export { default as LogosContainer } from './LogosContainer';
export { default as MetadataFieldsContainer } from './MetadataFieldsContainer';
export { default as PeopleAdminPage } from './PeopleAdminPage';
export { default as PermissionSearchFormContainer } from './PermissionSearchFormContainer';
export { default as ProjectSearchFormContainer } from './ProjectSearchFormContainer';
export { default as RegistryNameTypesSearchFormContainer } from './RegistryNameTypesSearchFormContainer';
export { default as RegistryReferenceTypesSearchFormContainer } from './RegistryReferenceTypesSearchFormContainer';
export { default as RoleSearchFormContainer } from './RoleSearchFormContainer';
export { default as SponsorLogosContainer } from './SponsorLogosContainer';
export { default as TaskTypeSearchFormContainer } from './TaskTypeSearchFormContainer';
export { default as UploadsPage } from './UploadsPage';
export { default as WrappedCollectionsContainer } from './WrappedCollectionsContainer';
export { default as WrappedContributionTypesContainer } from './WrappedContributionTypesContainer';
export { default as WrappedInstitutionsContainer } from './WrappedInstitutionsContainer';
export { default as WrappedLanguagesContainer } from './WrappedLanguagesContainer';
export { default as WrappedPermissionsContainer } from './WrappedPermissionsContainer';
export { default as WrappedProjectsContainer } from './WrappedProjectsContainer';
export { default as WrappedRegistryNameTypesContainer } from './WrappedRegistryNameTypesContainer';
export { default as WrappedRegistryReferenceTypesContainer } from './WrappedRegistryReferenceTypesContainer';
export { default as WrappedRolesContainer } from './WrappedRolesContainer';
export { default as WrappedTaskTypesContainer } from './WrappedTaskTypesContainer';
export { default as parametrizedQuery } from './parametrizedQuery';
