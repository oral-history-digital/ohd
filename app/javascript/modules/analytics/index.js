export { default as useTrackPageView } from './useTrackPageView';
export { default as useTrackPageViewFunction } from './useTrackPageViewFunction';
export { default as useTrackSiteSearch } from './useTrackSiteSearch';
export { default as useTrackEventFunction } from './useTrackEventFunction';
export { default as AnalyticsProvider } from './AnalyticsProvider';

export {
    EVENT_CATEGORY_WORKBOOK,
    SAVE_SEARCH,
    BOOKMARK_INTERVIEW,
    BOOKMARK_SEGMENT,
    EDIT_WORKBOOK_ITEM,
    REMOVE_WORKBOOK_ITEM,
} from './constants';
