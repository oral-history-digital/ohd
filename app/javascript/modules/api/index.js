export { default as Loader } from './Loader';
export { default as fetcher } from './fetcher';
export { default as apiMiddleware, CALL_API } from './middleware';
export { default as submitDataWithFetch } from './submitDataWithFetch';
export { default as useRegistryReferenceApi } from './useRegistryReferenceApi';
export { default as usePersonApi } from './usePersonApi';
export { default as useWorkbookApi } from './useWorkbookApi';
export { default as useDataApi } from './useDataApi';
export { default as useEventTypeApi } from './useEventTypeApi';
export { default as useHelpTextApi } from './useHelpTextApi';
