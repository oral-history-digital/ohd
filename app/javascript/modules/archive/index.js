export { NAME as ARCHIVE_NAME } from './constants';

export { addRemoveArchiveId, addRemoveRegistryEntryId, changeToEditView, changeToInterviewEditView,
    fetchStaticContent, setArchiveId, setArchiveIds, setLocale, setProjectId,
    setAvailableViewModes, setViewMode, clearViewModes, submitSelectedArchiveIds } from './actions';

export { default as archiveReducer } from './reducer';

export { getArchiveId, getEditView, getInterviewEditView, getLocale, getProjectId,
    getSelectedRegistryEntryIds, getTranslations, getDoiResult, getCountryKeys,
    getSelectedArchiveIds, getViewModes, getViewMode } from './selectors';

export { default as useIsEditor } from './useIsEditor';
