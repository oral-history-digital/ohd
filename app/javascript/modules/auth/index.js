export { useAuthorization } from './authorization-hook';
export { useProjectAccessStatus } from './projectAccessStatus-hook';
export { default as admin } from './admin';
export { default as AuthorizedContent } from './AuthorizedContent';
export { default as AuthShowContainer } from './AuthShowContainer';
export { PROJECT_ACCESS_GRANTED, PROJECT_ACCESS_REQUESTED } from './constants';
