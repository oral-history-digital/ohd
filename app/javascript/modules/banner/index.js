export { NAME as BANNER_NAME } from './constants';

export { default as Banner } from './Banner';
export { hideBanner } from './actions';
export { default as bannerReducer } from './reducer';
export { getBanner, getBannerActive } from './selectors';
export { bannerHasNotBeenHiddenByUser, doNotShowBannerAgainThisSession } from './bannerFunctions';
