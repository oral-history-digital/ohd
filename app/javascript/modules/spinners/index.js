export { default as Spinner } from './components/Spinner';
export { default as PixelLoader } from './components/PixelLoader';
export { default as MapSpinnerOverlay } from './components/MapSpinnerOverlay';
export { default as SearchSpinnerOverlay } from './components/SearchSpinnerOverlay';
