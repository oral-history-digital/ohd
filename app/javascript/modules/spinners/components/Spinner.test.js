import renderer from 'react-test-renderer';

import { Spinner } from 'modules/spinners';

it('renders correctly', () => {
    const tree = renderer
        .create(
            <Spinner
                className="my-spinner"
                style={{ margin: '1rem' }}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders padding CSS class', () => {
    const tree = renderer
        .create(
            <Spinner withPadding />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
