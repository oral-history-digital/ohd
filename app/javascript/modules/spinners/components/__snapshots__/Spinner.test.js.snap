// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `"<img src="mock.jpg" class="Spinner my-spinner" style="margin:1rem" alt=""/>"`;

exports[`renders padding CSS class 1`] = `"<img src="mock.jpg" class="Spinner Spinner--withPadding" alt=""/>"`;
